tests/CommunityToolkit.Mvvm.SourceGenerators.Roslyn401.UnitTests/Test_AsyncVoidReturningRelayCommandMethodCodeFixer.cs

```diff
@@ -0,0 +1,113 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading.Tasks;
+using CommunityToolkit.Mvvm.Input;
+using Microsoft.CodeAnalysis.Testing;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using CSharpCodeFixTest = Microsoft.CodeAnalysis.CSharp.Testing.CSharpCodeFixTest<
+    CommunityToolkit.Mvvm.SourceGenerators.AsyncVoidReturningRelayCommandMethodAnalyzer,
+    CommunityToolkit.Mvvm.CodeFixers.AsyncVoidReturningRelayCommandMethodCodeFixer,
+    Microsoft.CodeAnalysis.Testing.Verifiers.MSTestVerifier>;
+using CSharpCodeFixVerifier = Microsoft.CodeAnalysis.CSharp.Testing.CSharpCodeFixVerifier<
+    CommunityToolkit.Mvvm.SourceGenerators.AsyncVoidReturningRelayCommandMethodAnalyzer,
+    CommunityToolkit.Mvvm.CodeFixers.AsyncVoidReturningRelayCommandMethodCodeFixer,
+    Microsoft.CodeAnalysis.Testing.Verifiers.MSTestVerifier>;
+
+namespace CommunityToolkit.Mvvm.SourceGenerators.Roslyn401.UnitTests;
+
+[TestClass]
+public class Test_AsyncVoidReturningRelayCommandMethodCodeFixer
+{
+    [TestMethod]
+    public async Task AsyncVoidMethod_FileContainsSystemThreadingTasksUsingDirective()
+    {
+        string original = """
+            using System.Threading.Tasks;
+            using CommunityToolkit.Mvvm.Input;
+
+            partial class C
+            {
+                [RelayCommand]
+                private async void Foo()
+                {
+                }
+            }
+            """;
+
+        string @fixed = """
+            using System.Threading.Tasks;
+            using CommunityToolkit.Mvvm.Input;
+
+            partial class C
+            {
+                [RelayCommand]
+                private async Task Foo()
+                {
+                }
+            }
+            """;
+
+        CSharpCodeFixTest test = new()
+        {
+            TestCode = original,
+            FixedCode = @fixed,
+            ReferenceAssemblies = ReferenceAssemblies.Net.Net60
+        };
+
+        test.TestState.AdditionalReferences.Add(typeof(RelayCommand).Assembly);
+        test.ExpectedDiagnostics.AddRange(new[]
+        {
+            // /0/Test0.cs(7,24): error MVVMTK0039: The method C.Foo() annotated with [RelayCommand] is async void (make sure to return a Task type instead)
+            CSharpCodeFixVerifier.Diagnostic().WithSpan(7, 24, 7, 27).WithArguments("C.Foo()")
```

Comment:
I always find it hard to work with line/column numbers like this. For this case, I'd use `WithLocation(0)` instead of `WithSpan(...)` and then wrap the span with `{|#0:` and `|}`

Reply:
I'd prefer to leave it like this for consistency with the other tests. The advantage of this approach is that you can just copy-paste the initial output of the unit test when you first run it, so it's super easy to configure the expected diagnostics. As in, you never write any of this yourself, you just leave it blank, run the tests, then copy the output here 😄